### ISSUE TITLE
Consider whether to add a submit member function to the static_thread_pool sender type.

### ISSUE THREAD

**Issue by rarutyun** [executors/issaquah_2016#539]
Adding submit member function to ``static_thread_pool`` sender type allows implementations to customize the behavior according to their needs, rather then relying on the default behavior of ``std::execution::submit``.

For example implementation of ``static_thread_pool`` may want to customize how ``sender`` works with *Allocator*

**Comment by ben-craig** [comment]
I believe the customization point doesn't work if submit is a free function.  The user's free function is supposed to be a free function, but it needs to fall back to a member function.  We don't have a member function right now though, so it would fail.

**Comment by LeeHowes** [comment]
I don't think there's any need for it to be provided. It's implementation-defined how the submit CPO finds the customisation for the static_thread_pool. We don't need to specify how that happens, or even if it happens, for a given implementation.

My implementation of static_thread_pool would probably want to just use the default submit, at which point I would probably create an indirection type, call std::execution::submit against *it* and let it fall back to submit, just to avoid copying the implementation into the method.

**Comment by rarutyun** [comment]
Seems I recalled what we discuss. I don't think that the customization point wouldn't work in that case. I reckon it's right the opposite. Adding ``submit`` member function to ``static_thread_pool`` sender type allows implementations to customize the behavior according to their needs, rather then relying on the default behavior of `std::execution::submit` global whatever (function, CPO, etc.). We discussed considering the customization of how ``sender`` works with *Allocator*

**Comment by LeeHowes** [comment]
Yes, but:
 * The submit member function is not necessary to trigger that customisation. It may not be the preferred way to do so.
 * It is implementation-dependent whether it should be customised at all. The author may not wish to customise it at all.

So it is not necessary to require it and at worst it would be a burden if customisation was not required, leading to what I described.

**Comment by rarutyun** [comment]
Agree, but it may be stated as optional member. As we do for the *Allocator*. For example "if this method is provided the customization mechanism will find and call it". Otherwise, the default behavior is...".

That's what basically specified for the ``allocator_traits`` right now.

That's why we ask authors to investigate that direction.

**Comment by jaredhoberock** [comment]
To make this discussion more concrete, I think we are considering whether or not it is necessary to add a `submit` member to `static_thread_pool::scheduler_type::sender_type`'s specification in 2.5.4.
